api/src/main/java/org/xnio/ByteBufferSlicePool.java

```diff
@@ -206,8 +206,11 @@ public void clean() {
         if(!sliceQueue.isEmpty()) {
             sliceQueue.clear();
         }
-        // pass everything that is directly allocated to free direct buffers
-        FREE_DIRECT_BUFFERS.addAll(directBuffers);
+        // only true if using direct allocation
+        if (directBuffers != null) {
+            // pass everything that is directly allocated to free direct buffers
+            FREE_DIRECT_BUFFERS.addAll(directBuffers);
+        }
```

Comment:
By the way, as far as I read the code, nothing is added to `directBuffers` after the PR #214. So, I would like to ask how this `FREE_DIRECT_BUFFERS` works.

Reply:
Thanks for pointing that out, @msfm, I've just created XNIO-374